### DISCUSSION TITLE
Why is pip Package Nvidia "triton" in vllm's dependencies?

### DISCUSSION THREAD

**Discussion by jannikstdl** [vllm-project/vllm#10780] [Q&A]
Anyone knows for what this is needed? 
As far as i know triton supports vllm but not the other way around-

Installing collected packages: sentencepiece, pytz, pyairports, py-cpuinfo, nvidia-ml-py, mpmath, zipp, xxhash, websockets, uvloop, urllib3, tzdata, typing-extensions, tqdm, sympy, sniffio, six, safetensors, rpds-py, regex, pyzmq, pyyaml, python-dotenv, pycountry, pyarrow, psutil, protobuf, propcache, prometheus-client, pillow, partial-json-parser, packaging, nvidia-nvtx-cu12, nvidia-nvjitlink-cu12, nvidia-nccl-cu12, nvidia-curand-cu12, nvidia-cufft-cu12, nvidia-cuda-runtime-cu12, nvidia-cuda-nvrtc-cu12, nvidia-cuda-cupti-cu12, nvidia-cublas-cu12, numpy, networkx, nest-asyncio, msgspec, msgpack, MarkupSafe, llvmlite, lark, jiter, interegular, idna, httptools, h11, fsspec, frozenlist, filelock, exceptiongroup, einops, distro, diskcache, dill, cloudpickle, click, charset-normalizer, certifi, attrs, async-timeout, annotated-types, aiohappyeyeballs, uvicorn, 
**triton**
, requests, referencing, python-dateutil, pydantic-core, opencv-python-headless, nvidia-cusparse-cu12, nvidia-cudnn-cu12, numba, multiprocess, multidict, jinja2, importlib-metadata, httpcore, gguf, anyio, aiosignal, yarl, watchfiles, tiktoken, starlette, pydantic, pandas, nvidia-cusolver-cu12, jsonschema-specifications, huggingface-hub, httpx, torch, tokenizers, prometheus-fastapi-instrumentator, openai, lm-format-enforcer, jsonschema, fastapi, aiohttp, xformers, transformers, torchvision, ray, mistral-common, datasets, compressed-tensors, outlines, vllm

**Comment by jeejeelee** [comment]
If I understand your question correctly, the Triton you're referring to is https://github.com/triton-lang/triton
